File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def height(node):
            if not node:
                return 0
            left = height(node.left)
            right = height(node.right)
            return max(left,right)+1
        def balanced(node):
            if not node:
                return True
            if abs(height(node.left)-height(node.right))>1:
                return False
            left = balanced(node.left)
            right = balanced(node.right)
            return left and right

        return balanced(root)
        



            

        

        

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function isBalanced(root: TreeNode | null): boolean {
    let casee = 0;

    function balance(node){
      if(!node) return 1;
      const d1 = balance(node.left) + 1;
      const d2 = balance(node.right) + 1;

      if(Math.abs(d1 - d2) > 1) {
        casee++
      }

      return Math.max(d1,d2);
    }

    balance(root)

    return !Boolean(casee);
};